R/app_server.R

```diff
@@ -125,15 +125,114 @@ app_server <- function(input, output, session) {
+                 dplyr::select(-dplyr::any_of(getOption("emdash.cols_to_remove_from_participts_table"))) %>%
+                 data.table::setnames(originalColumnNames, new_column_names, skip_absent = TRUE)
     )
     callModule(mod_DT_server, "DT_ui_trips",
-      data = data_r$trips %>%
-        dplyr::select(-dplyr::any_of(getOption("emdash.cols_to_remove_trips_table"))) %>%
-        sf::st_drop_geometry()
+               data = data_r$trips %>%
+                 dplyr::select(-dplyr::any_of(getOption("emdash.cols_to_remove_trips_table"))) %>%
+                 sf::st_drop_geometry()
     )
+    
+    tableList <- getOption('emdash.supplementary_tables')
+    
+    # For each supplementary table, append a new tabPanel and run the server function
+    # that specifies table behavior
+    for (t in tableList){
+      table_type <- names(t)
+      table_title <- t[[table_type]]$tab_name
+      
+      # If the table is Bike Check In, use dtedit
+      if (table_type == 'Checkinout') {
+        
+        # Define the callback functions used by dtedit
+        #' Insert a row. "Create"
+        #' @param data the data including your inserted row
+        #' @param row the row where you made the change
+        insert_callback <- function(data, row) {
+          db_insert(cons,"Checkinout",data[row,])
+          return(data)
+        }
+        
+        #' Update a row
+        #' @param data the data including your updated row
+        update_callback <- function(data, olddata, row) {
+          db_update(cons,"Checkinout",data[row,])
+          data
+        }
+        
+        #' Delete a row
+        delete_callback <- function(data, row) {
+          db_delete(cons,"Checkinout",data[row,])
+          return(data[-row, ])
+        }
+        
+        # Make status a list so you can set false as one of the options in the event of all trues
+        data_for_dtedit <- data_r[[table_type]]
+        data_for_dtedit$status <- as.factor(data_for_dtedit$status)
+        
+        editable_table_tab <-  tabPanel(
+          status = "primary",
+          title = table_title,
+          value = table_type,
+          uiOutput(outputId = paste0("DTedit_ui_",table_type))  # Maybe Later: mod_DTedit_ui(id = paste0("DTedit_ui_",table_type)) #
+        )
+        
+        appendTab(
+          inputId = 'tabs',
+          tab = editable_table_tab,
+          select = FALSE,
+          menuName = NULL,
+          session = getDefaultReactiveDomain()
+        )
+
+        DTedit::dtedit(input, output,
+                       name = paste0("DTedit_ui_",table_type),
+                       thedata = data_for_dtedit,
+                       edit.cols = c('status'),
+                       edit.label.cols = c('status'),
+                       input.types = c(status = "selectInput"),
+                       input.choices = list(status = c('TRUE','FALSE')),
+                       view.cols = c('user_id','status', 'bikeLabel','ts','Checkout_time'),
```

Comment:
again, can the cols be determined from the input data instead of being hardcoded?

Reply:
I agree, I think Michael was just in a rush to finish this. Now that he is back he shall revise this. :)